File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed.
    int cents = get_cents();

    // Calculate the number of quarters to give the customer.
    int quarters = calculate_quarters(cents);
    cents -= quarters * 25;

    // Calculate the number of dimes to give the customer.
    int dimes = calculate_dimes(cents);
    cents -= dimes * 10;

    // Calculate the number of nickels to give the customer.
    int nickels = calculate_nickels(cents);
    cents -= nickels * 5;

    // Calculate the number of pennies to give the customer.
    int pennies = calculate_pennies(cents);
    cents -= pennies * 1;

    // Sum coins.
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer.
    printf("%d\n", coins);

    return 0;
}

int get_cents(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    } while (cents < 0);
    
    return cents;
}

int calculate_quarters(int cents)
{
    int quarters = cents / 25;

    return quarters;
}

int calculate_dimes(int cents)
{
    int dimes = cents / 10;

    return dimes;
}

int calculate_nickels(int cents)
{
    int nickels = cents / 5;

    return nickels;
}

int calculate_pennies(int cents)
{
    return cents;
}

File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    change = get_change()
    pennies = calculate_pennies(change)
    quarters = calculate_quarters(pennies)
    pennies -= quarters * 25
    dimes = calculate_dimes(pennies)
    pennies -= dimes * 10
    nickels = calculate_nickels(pennies)
    pennies -= nickels * 5
    coins = int(quarters + dimes + nickels + pennies)
    print(coins)


def get_change():
    change = get_float('Change owed: ')

    while change < 0:
        change = get_float('Change owed: ')
    
    return change


def calculate_pennies(change):
    pennies = change * 100

    return pennies


def calculate_quarters(cents):
    quarters = cents // 25

    return quarters


def calculate_dimes(cents):
    dimes = cents // 10

    return dimes


def calculate_nickels(cents):
    nickels = cents // 5

    return nickels


if __name__ == '__main__':
    main()